joomla/joomla-cms
# [4.0] Searchtool.js broken when clicking on table header an option which isn't in dropdown.

When you use SearchTools and have a table header sorting option which isn't in the sorting dropdown, it will not work in 4.0. There is a JS error `Cannot read property 'length' of undefined`.
This PR fixes that. @dgt41 As this is your code, can you check if the fix is correct? JS isn't my strongest area :smile: 

### Summary of Changes
* Checks if `desiredOption` exists before reading its property
* Change \JHtmlSearchtools::main to use minified file
* Change \JHtmlSearchtools::form to use the same call as ::main (was still using deprecated call)


### Testing Instructions
* Adjust the filter file (eg administrator/components/com_content/models/forms/filter_articles.xml) and remove some options from the `fullordering` field
* Make sure sorting still works in extensions using Smart Search (eg com_content)

### Expected result
All header sorting options work


### Actual result
Header sorting options only work if they also appear in the drodwon select.


### Documentation Changes Required
None
